bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/persistence/PersistenceResource.java

```diff
@@ -232,10 +313,160 @@ private Response getItemHistoryBean(String serviceName, String itemName, String
         for (Map.Entry<String, PersistenceService> service : persistenceServices.entrySet()) {
             ServiceBean serviceBean = new ServiceBean();
             serviceBean.name = service.getKey();
+            PersistenceService persistence = service.getValue();
+            if (persistence instanceof ModifiablePersistenceService) {
+                serviceBean.classname = "ModifiablePersistenceService";
+            } else if (persistence instanceof QueryablePersistenceService) {
+                serviceBean.classname = "QueryablePersistenceService";
+            } else {
+                serviceBean.classname = "PersistenceService";
+            }
 
             beanList.add(serviceBean);
         }
 
         return beanList;
     }
+
+    private Response getServiceItemList(String serviceName) {
+        // If serviceName is null, then use the default service
+        PersistenceService service = null;
+        if (serviceName == null) {
+            // TODO: Add handler for default service once this is available in ESH
```

Comment:
This doesn't seem to work. I added the managed service to the persistence resource - it gets called, but always with null parameter. I don't get the parameters with the default persistence being set until I remove the pid from the persistence model.

Any thoughts? We could add a ```PersistenceExtensions.getDefaultService()``` method, but that's another dependancy...